homeassistant/components/vlc_telnet/media_player.py

```diff
@@ -0,0 +1,231 @@
+"""Provide functionality to interact with the vlc telnet interface."""
+import logging
+import voluptuous as vol
+
+from homeassistant.components.media_player import (
+    MediaPlayerDevice, PLATFORM_SCHEMA)
+from homeassistant.components.media_player.const import (
+    MEDIA_TYPE_MUSIC, SUPPORT_PAUSE, SUPPORT_PLAY,
+    SUPPORT_PLAY_MEDIA, SUPPORT_STOP, SUPPORT_VOLUME_MUTE,
+    SUPPORT_VOLUME_SET, SUPPORT_PREVIOUS_TRACK, SUPPORT_SEEK,
+    SUPPORT_NEXT_TRACK, SUPPORT_CLEAR_PLAYLIST, SUPPORT_SHUFFLE_SET)
+from homeassistant.const import (
+    CONF_NAME, STATE_IDLE, STATE_PAUSED, STATE_PLAYING, STATE_UNAVAILABLE,
+    CONF_HOST, CONF_PORT, CONF_PASSWORD)
+import homeassistant.helpers.config_validation as cv
+
+_LOGGER = logging.getLogger(__name__)
+
+DOMAIN = 'vlc_telnet'
+
+DEFAULT_NAME = 'Vlc Telnet'
+
+SUPPORT_VLC = SUPPORT_PAUSE | SUPPORT_SEEK | SUPPORT_VOLUME_SET \
+              | SUPPORT_VOLUME_MUTE | SUPPORT_PREVIOUS_TRACK \
+              | SUPPORT_NEXT_TRACK | SUPPORT_PLAY_MEDIA | SUPPORT_STOP \
+              | SUPPORT_CLEAR_PLAYLIST | SUPPORT_PLAY \
+              | SUPPORT_SHUFFLE_SET
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Optional(CONF_HOST, default='127.0.0.1'): cv.string,
+    vol.Optional(CONF_PORT, default='4212'): cv.positive_int,
+    vol.Required(CONF_PASSWORD, default='test'): cv.string,
+    vol.Optional(CONF_NAME, default='VLC-TELNET'): cv.string,
+})
+
+
+def setup_platform(hass, config, add_entities, discovery_info=None):
+    """Set up the vlc platform."""
+    add_entities([VlcDevice(config.get(CONF_NAME),
+                            config.get(CONF_HOST),
+                            config.get(CONF_PORT),
+                            config.get(CONF_PASSWORD))])
+
+
+class VlcDevice(MediaPlayerDevice):
+    """Representation of a vlc player."""
+
+    def __init__(self, name, host, port, passwd):
+        """Initialize the vlc device."""
+        self._instance = None
+        self._name = name
+        self._volume = None
+        self._muted = None
+        self._state = STATE_UNAVAILABLE
+        self._media_position_updated_at = None
+        self._media_position = None
+        self._media_duration = None
+        self._host = host
+        self._port = port
+        self._password = passwd
+        self._vlc = None
+        self._volume_bkp = 0
+        self._media_artist = ""
+        self._media_title = ""
+        print("loaded!")
+
+    def update(self):
+        """Get the latest details from the device."""
+        from python_telnet_vlc import VLCTelnet, ConnectionError as ConnErr
+
+        if self._vlc is None:
+            try:
+                self._vlc = VLCTelnet(self._host, self._password, self._port)
+                self._state = STATE_IDLE
+            except ConnErr:
+                self._state = STATE_UNAVAILABLE
```

Comment:
Instead of overriding state, implement `def available` and return True or False.

Reply:
Mmm I didn't understand this one. "Available" is not a method to override on the MediaPlayerDevice super class. Instead, "def state" is, which uses the value self._state to return the current state.

Reply:
```
def ini...
   self._available = False # (or true what you like)

@property
def available(self):
   """..COPY ME FROM helpers/entity.py"""
  return self._available

```